src/DynamoCoreWpf/Utilities/WebView2Utilities.cs

```diff
@@ -15,31 +19,53 @@ public class DynamoWebView2 : WebView2
         private string tag;
         #endregion
 
+        Action<string> logger = System.Console.WriteLine;
+        private Task initTask = null;
+        private bool diposeCalled;
+
         public DynamoWebView2() : base()
         {
             tag = TestUtilities.WebView2Tag;
         }
 
+        /// <summary>
+        /// Wrapper over WebView2's EnsureCoreWebView2Async
+        /// Use this method instead of EnsureCoreWebView2Async
+        /// </summary>
+        /// <returns></returns>
+        internal async Task Initialize(Action<string> logFn = null)
+        {
+            logger ??= logFn;
+            initTask = EnsureCoreWebView2Async();
+            await initTask;
+
+            ObjectDisposedException.ThrowIf(diposeCalled, this);
```

Comment:
so this is helpful to trigger early failure?

Reply:
Yes, it forces all callers to handle the exception. 
Ex 
```
await webview2.Initialize(); \\ - will throw Disposed exception if it was disposed before execution resumes
\\ - webview2.Dispose could be called before this function resumes execution
webview2.DoStuff(); \\ - Do not execute any more code if Initialize throws an exception
```